### PR TITLE
Update tslint react to v2.5.0

### PR DESCRIPTION
**Pull request by ismail-syed**
Update tslint-react to [v2.5.0](https://github.com/palantir/tslint-react/releases)

* Removed our custom rule for `jsx-boolean-value`
* Added `jsx-use-translation-function` and `jsx-boolean-value` from `tslint-react`

### CODE REVIEW

File: src/rules/react.ts [hunk 1]
```diff
@@ -8,6 +8,8 @@ export default {
   'jsx-alignment': true,
   // Allows blacklisting of props in JSX with an optional explanatory message in the reported failure.
   'jsx-ban-props': false,
+  // When using a boolean attribute in JSX, you can set the attribute value to true or omit the value. // This rule will enforce one or the other to keep consistency in your code.
+  'jsx-boolean-value': true,
```
Comment (ismail-syed) [inline src/rules/react.ts]: We ended up changing the default to be `always`. The `tslint-react` maintainer preferred it over `never`.
[See PR](https://github.com/palantir/tslint-react/pull/70#issuecomment-285798353)

File: src/rules/react.ts [hunk 2]
```diff
@@ -20,6 +22,8 @@ export default {
   'jsx-no-string-ref': true,
   // Enforces that JSX elements with no children are self-closing.
   'jsx-self-close': true,
+  // Enforces use of a translation function. Plain string literals are disallowed in JSX when enabled.
+  'jsx-use-translation-function': true,
```
Comment (GoodForOneFare) [inline src/rules/react.ts:26]: 💯 Might take a while to get Neutron working with this, but our users will love us for it.

File: src/rules/react.ts (hunk 1 above)
Comment (GoodForOneFare) [reply src/rules/react.ts]: But we prefer `never`, right?  I think we need to make that explicit here.

**Review by GoodForOneFare — APPROVED** [review]
That's a beauty.  One comment that I could be wrong about.  Either way, 🚢 soon.

File: src/rules/react.ts (hunk 1 above)
Comment (ismail-syed) [reply src/rules/react.ts]: Yes, you're right. I keep getting those mixed up.

**Comment by ismail-syed** [comment]
@GoodForOneFare Fixes are up! 👍

**Review by GoodForOneFare — APPROVED** [review]

**Pull request merged**